core/templates/components/summary-tile/lesson-card.component.ts

```diff
@@ -0,0 +1,163 @@
+    }
+  }
+
+  setStorySummary(storyModel: StorySummary): void {
+    const completedStories = storyModel.getCompletedNodeTitles().length;
+    this.desc = storyModel.getTitle();
+    this.imgColor = storyModel.getThumbnailBgColor();
+    this.imgUrl = this.getStorySummaryThumbnailUrl({
+      filename: storyModel.getThumbnailFilename(),
+      id: storyModel.getId(),
+    });
+
+    const nextStory =
+      completedStories === storyModel.getAllNodes().length
+        ? completedStories - 1
+        : completedStories;
+
+    const lessonArgs = {
+      classroom: storyModel.getClassroomUrlFragment(),
+      topicFragment: storyModel.getTopicUrlFragment(),
+      currentStory: storyModel.getAllNodes()[nextStory],
+      story: storyModel.getUrlFragment(),
+    };
+    this.lessonUrl = this.getStorySummaryLessonUrl(lessonArgs);
+
+    this.title = `Chapter ${nextStory + 1}: ${storyModel.getNodeTitles()[nextStory]}`;
+    this.progress = Math.floor(
+      (completedStories / storyModel.getNodeTitles().length) * 100
+    );
+    this.lessonTopic = this.topic;
+  }
+
+  setCollectionSummary(collectionModel: CollectionSummary): void {
+    this.desc = collectionModel.objective;
+    this.imgColor = collectionModel.thumbnailBgColor;
+    this.imgUrl = this.urlInterpolationService.getStaticImageUrl(
+      collectionModel.thumbnailIconUrl
+    );
+    this.progress = 0;
+    this.title = collectionModel.title;
+    this.lessonUrl = `/collection/${collectionModel.id}`;
+    this.lessonTopic = 'Collections';
+  }
+
+  setExplorationSummary(explorationModel: LearnerExplorationSummary): void {
+    this.desc = explorationModel.objective;
+    this.imgColor = explorationModel.thumbnailBgColor;
+    this.imgUrl = this.urlInterpolationService.getStaticImageUrl(
+      explorationModel.thumbnailIconUrl
+    );
+    this.progress = 0;
+    this.title = explorationModel.title;
+    this.lessonUrl = `/explore/${explorationModel.id}`;
+    this.lessonTopic = 'Community Lessons';
+  }
+
+  getStorySummaryThumbnailUrl(storyData: {
+    filename: string;
+    id: string;
+  }): string {
+    return this.assetsBackendApiService.getThumbnailUrlForPreview(
+      AppConstants.ENTITY_TYPE.STORY,
+      storyData.id,
+      storyData.filename
+    );
+  }
+
+  getStorySummaryLessonUrl(storyData: {
+    classroom: string | undefined;
+    topicFragment: string | undefined;
+    story: string;
+    currentStory: StoryNode;
+  }): string {
+    if (!storyData.classroom || !storyData.topicFragment) {
+      throw new Error('Class and/or topic does not exist');
+    }
+    return (
+      `/explore/${storyData.currentStory.getExplorationId()}?` +
+      Object.entries({
+        topic_url_fragment: storyData.topicFragment,
+        classroom_url_fragment: storyData.classroom,
+        story_url_fragment: storyData.story,
+        node_id: storyData.currentStory.getId(),
+      })
+        .map(([key, value]) => `${key}=${value}`)
+        .join('&')
+    );
```

Comment:
That's a very complicated way to construct a url. I think constructing it like
`topic_url_fragment=${topicFragment}&...` would be more readable

Reply:
Done.

Reply:
@amyyeung17 FYI we have a UrlInterpolationService that you might find useful.

Reply:
@seanlip Ah, should I create a new URL template to to use with UrlnterpolationService then? The previous iteration just called urlService.addField a bunch of times, which I wanted to avoid.

Reply:
> @seanlip Ah, should I create a new URL template to to use with UrlnterpolationService then? The previous iteration just called urlService.addField a bunch of times, which I wanted to avoid.

Yup. Confirming what @nikitaevg said, please use UrlInterpolationService for handling URL construction, thanks! (This is how we should do it across the board.)